airflow/provider.yaml.schema.json

```diff
@@ -167,6 +170,13 @@
           "python-module"
         ]
       }
+    },
+    "hook-class-names": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "description": "Hooks class names that provide connection types to core"
```

Comment:
```suggestion
      "type": "array",
      "description": "Hooks class names that provide connection types to core"
      "items": {
        "type": "string",
```
If we describe arrays, we can use the plural in the sentence. If we describe an item, we should use singular form in the description.

Reply:
What's the difference between the two in terms of UI/validation? Should we describe item or array and what happens if we do both?

Reply:
It seems to me that we should try to describe the most general fields. If we need it, we can also describe lower level elements.

From a validation perspective, these fields are ignored, but describing the list of items provides better hints in the IDE.
Before:
<img width="609" alt="Screenshot 2020-11-28 at 18 11 53" src="https://user-images.githubusercontent.com/12058428/100521656-44e2a100-31a5-11eb-9b84-1cc55bc016e9.png">

After:
<img width="639" alt="Screenshot 2020-11-28 at 18 10 02" src="https://user-images.githubusercontent.com/12058428/100521645-3a280c00-31a5-11eb-8412-efd798a608e9.png">

Reply:
OK. I went ahead fixed a few typos and added some missing descriptions in #12690 as the behaviour and descriptions were not consistent in the schema.